packages/fuels-types/src/param_types.rs

```diff
@@ -67,6 +68,44 @@ impl ParamType {
         }
     }
 
+    pub fn contains_nested_vectors(&self) -> bool {
+        match &self {
+            ParamType::Vector(param_type) => param_type.uses_vectors(),
+            _ => self.uses_vectors(),
+        }
+    }
+
+    fn uses_vectors(&self) -> bool {
+        match &self {
+            ParamType::Vector(..) => true,
+            ParamType::Array(param_type, ..) => param_type.uses_vectors(),
+            ParamType::Tuple(param_types, ..) => Self::any_nested_vectors(param_types),
+            ParamType::Enum {
+                generics, variants, ..
+            } => {
+                let variants_types = variants.param_types();
+                Self::any_nested_vectors(chain!(generics, &variants_types))
+            }
+            ParamType::Struct {
+                fields, generics, ..
+            } => {
+                let fields = fields.iter().map(|(_, param_type)| param_type);
+                Self::any_nested_vectors(chain!(fields, generics))
+            }
+            _ => false,
+        }
+    }
+
+    fn any_nested_vectors<'a>(param_types: impl IntoIterator<Item = &'a ParamType>) -> bool {
+        param_types
+            .into_iter()
+            .any(|param_type| param_type.is_vector() || param_type.contains_nested_vectors())
```

Comment:
We're allowing for one vector as the root type in `contains_nested_vectors`. Any further check we do should only detect whether vectors are used. If they are that makes them nested vectors.
```suggestion
            .any(|param_type| param_type.uses_vectors())
```